include/knowhere/utils.h

```diff
@@ -0,0 +1,30 @@
+// Copyright (C) 2019-2023 Zilliz. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under the License.
+
+#pragma once
+
+#include "knowhere/config.h"
+#include "knowhere/dataset.h"
+
+namespace knowhere {
+
+inline bool
+IsMetricType(const std::string& str, const knowhere::MetricType& metric_type) {
```

Comment:
To my understanding, only `brute_force` need string comparison for metrics and all others can directly use enum matching right?

Can we have a new API to compare enum class instead of turning them to string.

Reply:
only after metric string converted to faiss::metric, we can use enum matching,
for HNSW and DISKANN, we have to use string compare.